### PR TITLE
Merging Lifecycle Callbacks code samples for PHP + XML + YAML

### PR DESCRIPTION
**Pull request by ThomasLandauer**
IMO, the text I deleted just repeated things that are obvious in the example anyway.

I didn't change anything in the three code blocks - just merged them. GitHub Diff looks a little weird...

### CODE REVIEW

File: docs/en/reference/events.rst [hunk 1]
```diff
@@ -299,133 +299,89 @@ specific to a particular entity class's lifecycle.
...
-    # ...
-        name:
-          type: string(50)
-      lifecycleCallbacks:
-        prePersist: [ doStuffOnPrePersist, doOtherStuffOnPrePersist ]
-        postPersist: [ doStuffOnPostPersist ]
-
-In YAML the ``key`` of the lifecycleCallbacks entry is the event that you
-are triggering on and the value is the method (or methods) to call. The allowed
-event types are the ones listed in the previous Lifecycle Events section.
```
Comment (SenseException) [inline docs/en/reference/events.rst:371]: Can you please put the text below the code configuration block like you did with the annotation one?

File: docs/en/reference/events.rst [hunk 2]
```diff
@@ -299,133 +299,89 @@ specific to a particular entity class's lifecycle.
...
+            {
+                $this->value = 'changed from preUpdate callback!';
+            }
+        }
+    .. code-block:: xml
 
-When using YAML or XML you need to remember to create public methods to match the
-callback names you defined. E.g. in these examples ``doStuffOnPrePersist()``,
-``doOtherStuffOnPrePersist()`` and ``doStuffOnPostPersist()`` methods need to be
-defined on your ``User`` model.
```
Comment (SenseException) [inline docs/en/reference/events.rst]: Can you please put the text below the code configuration block like you did with the annotation one?

**Review by SenseException — CHANGES_REQUESTED** [review]

File: docs/en/reference/events.rst (hunk 1 above)
Comment (ThomasLandauer) [reply docs/en/reference/events.rst:371]: I wanted to take care of this in another PR ;-) - now included it in the second (third?) commit here.

The larger problem I'm seeing is that the three code blocks were showing *different* events ;-) I now shortened the PHP block and made all three consistent.

About the specific (deleted) text: I think the example shows it all! The name of the functions, and that they need to be public. So all that's left is `@HasLifecycleCallbacks`, and that I included in the code block.

Comment (SenseException) [reply docs/en/reference/events.rst:371]: >  I think the example shows it all!

Maybe it is self-explaining. for the code blocks. Okay, let's keep it that way.

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @ThomasLandauer !

**Pull request merged**